File: java/src/main/java/vsdk/toolkit/raycasting/bidirectionalRaytracing/DensityHit.java
package vsdk.toolkit.raycasting.bidirectionalRaytracing;

import vsdk.toolkit.common.ColorRgb;

/**
Class DensityBuffer : class for storing sample hits on screen
New samples are added with 'Add'. 'reconstruct' reconstructs
an approximation to the sampled function into a screen buffer
*/
public class DensityHit {
    public float m_x; // Screen/Polygon Coordinates
    public float m_y;
    public ColorRgb color; // Estimate of the function, NOT divided by number of samples

    public DensityHit() {
        color = new ColorRgb();
    }

    public void init(float x, float y, ColorRgb col) {
        m_x = x;
        m_y = y;
        color = new ColorRgb(col.r, col.g, col.b);
    }

    public DensityHit(float x, float y, ColorRgb col) {
        this();
        init(x, y, col);
    }
}

File: typescript/src/main/vsdk/toolkit/raycasting/bidirectionalRaytracing/DensityHit.ts
import { ColorRgb } from "../../common/color/ColorRgb";

export class DensityHit {
  public m_x: number;
  public m_y: number;
  public color: ColorRgb;

  public constructor();
  public constructor(x: number, y: number, col: ColorRgb);
  public constructor(x?: number, y?: number, col?: ColorRgb) {
    this.m_x = 0.0;
    this.m_y = 0.0;
    this.color = new ColorRgb();

    if (x !== undefined && y !== undefined && col !== undefined) {
      this.init(x, y, col);
    }
  }

  public init(x: number, y: number, col: ColorRgb): void {
    this.m_x = x;
    this.m_y = y;
    this.color = new ColorRgb(col.r, col.g, col.b);
  }
}
